sdk/src/wallet/storage/participation.rs

```diff
@@ -110,3 +110,57 @@ impl StorageManager {
             .unwrap_or_default())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{types::block::payload::transaction::TransactionId, wallet::storage::adapter::memory::Memory};
+
+    #[tokio::test]
+    async fn insert_get_remove_participation_event() {
+        let mut storage_manager = StorageManager::new(None, Box::<Memory>::default()).await.unwrap();
+        assert!(storage_manager.get_participation_events(0).await.unwrap().is_empty());
+
+        storage_manager
+            .insert_participation_event(0, ParticipationEventWithNodes::mock())
+            .await
+            .unwrap();
+        let events = storage_manager.get_participation_events(0).await.unwrap();
+        assert_eq!(events.len(), 1);
```

Comment:
Compare the result?

Reply:
Requires `ParticipationEventWithNodes` to be `PartialEq` (at least) which is possible since all its fields are `PartialEq`, so I think we should just add the derive and compare the result.

Reply:
Done